### PR TITLE
feat: add support for async methods 

### PR DESCRIPTION
**Pull request by leohoare**
Introduce async support using httpx.

### CODE REVIEW

File: confidence/openfeature_provider.py [hunk 1]
```diff
@@ -121,6 +121,24 @@ def resolve_boolean_details(
             flag_metadata=details.flag_metadata,
         )
 
+    async def resolve_boolean_details_async(
```
Comment (nicklasl) [inline confidence/openfeature_provider.py:124]: The OpenFeature provider should not be used on it's own but rather together with the [OpenFeature SDK](https://github.com/open-feature/python-sdk/) so I would revert the changes in this class (and the test).

**Review by nicklasl — COMMENTED** [review]
Thanks for the contributions! 
IMO the async API make sense, unfortunately I don't know too much about Python and if there are different libraries to use here?
For example, is `asyncio` an alternative approach? 
 
Other than that, there seems to be some opportunity to refactor to reuse some of the code in the `evalate` and `resolve` functions... I could live with that but the fruit is pretty low hanging.

File: confidence/openfeature_provider.py (hunk 1 above)
Comment (leohoare) [reply confidence/openfeature_provider.py]: Alright this has been removed, I'll look into adding the feature into OpenFeature SDK.

**Comment by leohoare** [comment]
> IMO the async API make sense, unfortunately I don't know too much about Python and if there are different libraries to use here?
For example, is asyncio an alternative approach?

httpx and aiohttp are built on top of asyncio and tend to be the go-to options for async requests due to their robustness. Implementing requests with asyncio directly tends to be pretty messy and error prone.
As for the choice between implementations:
* httpx is a drop in replacement for requests, supports sync/async (if you wanted to refactor to use a single library), more "modern" approach.
* aiohttp has a larger following and the standard for a relatively long time, a bit clunkier to implement.

This [article](https://apidog.com/blog/aiohttp-vs-httpx/) has a good comparison.
image

Something to consider is the default configuration on the client e.g. timeout default is 5 seconds with httpx.

> Other than that, there seems to be some opportunity to refactor to reuse some of the code in the evalate and resolve functions... I could live with that but the fruit is pretty low hanging.

Happy to do this change, I was initially avoid refactoring. It's a common issue in python is supporting sync/async you end up with large chunks of forked code.

File: confidence/openfeature_provider.py (hunk 1 above)
Comment (nicklasl) [reply confidence/openfeature_provider.py]: I asked about it in their slack yesterday and was pointed to [this issue](https://github.com/open-feature/python-sdk/issues/284). Federico that has commented on that issue is one of the main contributors to the OpenFeature Python SDK and he seems positive 👍
